### PR TITLE
docs: rekeying transaction

### PR DESCRIPTION
**Pull request by vuvth**
## Proposed Changes

+  Rekeying documents

### CODE REVIEW

File: docs/guide/runtime-advanced.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+layout: splash
+---
+
+# Rekeying transaction
+
+- You should read about Algorand rekeying, [here](https://developer.algorand.org/docs/get-details/accounts/rekey/).
```
Comment (robert-zaremba) [inline docs/guide/runtime-advanced.md:7]:
```suggestion
Read first about [Algorand rekeying](https://developer.algorand.org/docs/get-details/accounts/rekey/). 
```

File: docs/guide/runtime-advanced.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+---
+
+# Rekeying transaction
+
+- You should read about Algorand rekeying, [here](https://developer.algorand.org/docs/get-details/accounts/rekey/). 
+- `Runtime` supported a rekeying transaction. You can use field `rekeyTo` in `payFlags` to rekey from account to another address. 
+- We can rekeying:
+    - Account to any account.
+    - Logic account to any account.
+    - Application to any account. We use inner transaction to rekey application.
```
Comment (robert-zaremba) [inline docs/guide/runtime-advanced.md:12]:
```suggestion
`Runtime` supports a rekey transaction. You can use field `rekeyTo` in `payFlags` to rekey:
- Account to any account.
- Smart Signature account to any account.
- Application to any account. We use inner transaction to rekey application.
```

File: docs/guide/runtime-advanced.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+Example: 
+
+- We can rekey join account to bob follow below config.
```
Comment (robert-zaremba) [inline docs/guide/runtime-advanced.md:16]:
```suggestion
We can rekey John account to Bob follow suing the following transaction parms.
```

File: docs/guide/runtime-advanced.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    fromAccount: john,
+    toAccountAddr: alice.address,
+    amountMicroAlgos: 0n,
+    payFlags: { 
+        totalFee: fee 
+        rekeyTo: bob.address
+    } 
+```
+
+- We can rekey logic signature (lsig) to bob.
```
Comment (robert-zaremba) [inline docs/guide/runtime-advanced.md:29]:
```suggestion
We can rekey smart signature (lsig) to bob.
```

File: docs/guide/runtime-advanced.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    toAccountAddr: bob.address,
+    amountMicroAlgos: 0n,
+    payFlags: {
+        totalFee: fee 
+        rekeyTo: bob.address 
+    }
+}
+```
+
+You can check how to use rekeying transaction at  [here](https://github.com/scale-it/algo-builder/blob/master/packages/runtime/test/integration/rekey-transaction.ts).
```
Comment (robert-zaremba) [inline docs/guide/runtime-advanced.md:45]:
```suggestion
You can check how to use rekeying transaction in the [source code](https://github.com/scale-it/algo-builder/blob/master/packages/runtime/test/integration/rekey-transaction.ts).
```

File: packages/runtime/test/integration/rekey/rekey-and-inner-tx.ts [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+        }
+      };
+      runtime.executeTx(tranferAlgoTx);
+
+      syncAccounts();
+      const applBalanceAfter = appAccount.amount;
+      const aliceBalanceAfter = alice.amount;
+      const bobBalanceAfter = bob.amount;
+
+      // alice account not change
```
Comment (robert-zaremba) [inline packages/runtime/test/integration/rekey/rekey-and-inner-tx.ts:126]:
```suggestion
      // alice account didn't change
```

File: packages/runtime/test/integration/rekey/rekey-and-inner-tx.ts [hunk 7]
```diff
@@ -0,0 +1,246 @@
...
+        sign: types.SignType.SecretKey,
+        fromAccount: bob.account,
+        fromAccountAddr: appAccount.address,
+        toAccountAddr: bob.address,
+        amountMicroAlgos: amount,
+        payFlags: {
+          totalFee: fee
+        }
+      };
+      // should failed
```
Comment (robert-zaremba) [inline packages/runtime/test/integration/rekey/rekey-and-inner-tx.ts:146]:
```suggestion
      // should fail
```

**Review by robert-zaremba — APPROVED** [review]

File: docs/guide/runtime-advanced.md [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+
+- Read first about [Algorand rekeying](https://developer.algorand.org/docs/get-details/accounts/rekey/). 
+`Runtime` supports a rekey transaction. You can use field `rekeyTo` in `payFlags` to rekey:
+- Account to any account.
+- Smart Signature account to any account.
+- Application to any account. We use inner transaction to rekey application.
+
+Example: 
+
+We can rekey John account to Bob follow suing the following transaction params.
```
Comment (sczembor) [inline docs/guide/runtime-advanced.md:15]:
```suggestion
We can rekey John account to Bob using the following transaction params.
```

File: packages/runtime/test/integration/rekey/rekey-and-inner-tx.ts [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+        clearProgramFileName,
+        appDeployFlag,
+        {}
+      ).appID;
+
+      syncAccounts();
+      // query appAccount
+      appAccount = runtime.getAccount(getApplicationAddress(appID));
+
+      // fund app (escrow belonging to app) with 10 ALGO
```
Comment (sczembor) [inline packages/runtime/test/integration/rekey/rekey-and-inner-tx.ts:69]:
```suggestion
      // fund app (escrow belonging to app) with 1 ALGO
```

File: packages/runtime/test/integration/rekey/rekey-and-inner-tx.ts [hunk 10]
```diff
@@ -0,0 +1,246 @@
...
+      runtime.executeTx(tranferAlgoTx);
+
+      syncAccounts();
+      const applBalanceAfter = appAccount.amount;
+      const aliceBalanceAfter = alice.amount;
+      const bobBalanceAfter = bob.amount;
+
+      // alice account didn't change
+      assert.equal(aliceBalanceBefore, aliceBalanceAfter);
+      // include fee and amount use for inner transaction
```
Comment (sczembor) [inline packages/runtime/test/integration/rekey/rekey-and-inner-tx.ts:128]:
```suggestion
      // include fee and amount used for inner transaction
```

**Review by sczembor — COMMENTED** [review]
Few suggestions

**Pull request merged**